Website Documentation

awholegunch opened the pull request:
Website documentation, including Javadocs and additional materials.
Results are now staged at http://stg-web101.sw.joyent.continuuity.net/tigon/0.2.0-SNAPSHOT/en/

Commit:
Add copyright symbol "©" to copyright statements.

Commit:
Documentation framework. Modifies pom.xml to build javadocs; sets up initial files and build scripts for rst files for generating HTML.
Adds a preliminary placeholder Tigon logo.

Commit:
Merge branch 'develop' into feature/docs-guides

Commit:
Removing Continuuity.

Commit:
Tigon docs

Commit:
Architecture doc.

Commit:
Fix formatting and title levels.

Commit:
Fix link to javadocs.

Commit:
More docs.

Commit:
More updates.

Commit:
Revise opening.

Commit:
Revised architecture diagram and description in documentation.

Commit:
Remove more references to NICs.

Commit:
Updates to build script.

Commit:
Fixed broken link.

Commit:
Minor fixes.

Commit:
Revised logo.

Commit:
Fix IRC link.

Commit:
Cleaner version of the logo

Commit:
Add FAQ questions

Commit:
Re-organization of Tigon docs and build script.

Commit:
Update filename and speed up version check.

Commit:
FAQ and Tools (Command line interface)

Commit:
Updates to Tigon dcoumentation.

Commit:
Updates, and a fix for the full javadoc build.

Commit:
Merge branch 'develop' into feature/docs-guides

Conflicts:
	checkstyle.xml
	pom.xml
	suppressions.xml

Commit:
Fixes from merge.

Commit:
Updates to fix build of full javadocs.

Commit:
Adding Google Analytics tracking code.
Adding GitHub pages version for building.

Commit:
Updated Google Analytics codes.

Commit:
Fixes to build script.

Commit:
New Tigon SQL Contributor manual in PDF format...

Commit:
FAQ update.

Commit:
Add a quick build to the build script.
Added administrating a Tigon instance.
Added additional FAQ answers.
Corrected the command-line interface documentation.

Commit:
Fix broken indentation.

Commit:
Fix formatting.

Commit:
Update the logo for the doc set; remove the old one, standardize the CSS to use "doc_logo.png".

Commit:
Updates to the guides.

Commit:
Adding example, fix getting started.

Commit:
Remove external images

achalpandey commented on tigon-docs/developer-guide/source/admin.rst:
```diff
@@ -0,0 +1,71 @@
+Where to Go Next
+================
+
+Now that you're seen how to run Tigon, take a look at:
```
`Now that you "have" seen...`
Or possible rephrasing

achalpandey replied:
`Now that you're familiar with how to run Tigon`
Similar to https://github.com/caskdata/tigon/pull/9/files#diff-08155194f314a1c6706bd19e5a78f585R450

achalpandey commented on tigon-docs/developer-guide/source/apis/apis.rst:
```diff
@@ -0,0 +1,36 @@
+Where to Go Next
+================
+
+Now that you're seen the APIs for Tigon, take a look at:
```
Same as above.

achalpandey commented on tigon-docs/developer-guide/source/developer.rst:
```diff
@@ -0,0 +1,622 @@
+atomic and "exactly-once" processing of each input object by each
+Flowlet in the DAG.
+
+Flows are deployed to the CDAP instance and hosted within containers. Each
```
This needs to be changed. Tigon now runs standalone and isn't related with CDAP.

Suggested edit: `A Flow runs as a YARN application and each flowlet instance runs in its own container.`

achalpandey commented on tigon-docs/developer-guide/source/developer.rst:
```diff
@@ -0,0 +1,622 @@
+        .withFlowlets()
+          .add("flowlet1", new MyExampleFlowlet())
+          .add("flowlet2", new MyExampleFlowlet2())
+        .connect()
```
Please change to:
`.add("flowlet1", new MyExampleFlowlet(), 1)`
`.add("flowlet2", new MyExampleFlowlet2(), 3)`

achalpandey commented on tigon-docs/developer-guide/source/developer.rst:
```diff
@@ -0,0 +1,622 @@
+
+In this example, the *name*, *description*, *with* (or *without*)
+Flowlets, and *connections* are specified before building the Flow.
+
```
Can we please add a short explanation for the above flow.

Something like: `The above code creates a flow with two flowlets named "flowlet1", a MyExamplFlowlet() with 1 instance, and "flowlet2", a MyExampleFlowlet2() with 3 instances.`

Commit:
Update building.

Commit:
Updates to release notes.

Commit:
More updates for docs.

achalpandey commented on tigon-docs/developer-guide/source/developer.rst:
```diff
@@ -0,0 +1,622 @@
+Flowlets, and *connections* are specified before building the Flow.
+
+.. _flowlets:
+
```
@gokulavasan Do you think we should add a small note about flows explaining that a flow requires that there are at least 2 flowlets? As this information should be surfaced appropriately

gokulavasan replied:
@achalpandey @awholegunch Yes we should mention that a Flow needs to have two flowlets. Also should we add it known issues?

Commit:
Merge branch 'develop' into feature/docs-guides

gokulavasan commented on tigon-docs/developer-guide/source/architecture.rst:
```diff
@@ -0,0 +1,453 @@
+Tigon Concepts
+============================================
+
+Tigon Architecture
```
This should be renamed as TigonSQL Architecture

gokulavasan commented on tigon-docs/developer-guide/source/developer.rst:
```diff
@@ -0,0 +1,623 @@
+    .add(new RandomGenerator())
+    .add(new RoundingFlowlet())
+  .connect()
+    .fromStream("RandomGenerator").to("RoundingFlowlet")
```
No streams. We can connect only flowlets. .from("Rand..").to("Round..")

gokulavasan commented on tigon-docs/developer-guide/source/developer.rst:
```diff
@@ -0,0 +1,623 @@
+`REST interfaces <rest.html>`__ or via the CDAP Console. This enables you
+to scale your application to meet capacity at runtime.
+
+In the Local DAP, multiple Flowlet instances are run in threads, so in some cases
```
In Tigon Standalone, multiple Flowlet instances....

gokulavasan commented on tigon-docs/developer-guide/source/developer.rst:
```diff
@@ -0,0 +1,623 @@
+to scale your application to meet capacity at runtime.
+
+In the Local DAP, multiple Flowlet instances are run in threads, so in some cases
+actual performance may not be improved. However, in the Distributed DAP,
```
in Tigon Distributed

gokulavasan commented on tigon-docs/developer-guide/source/developer.rst:
```diff
@@ -0,0 +1,623 @@
+
+  public class Counter extends AbstractFlowlet {
+
+    @UseDataSet("wordCounts")
```
We need to remove this example. Or write something different since we don't have any datasets in Tigon.

gokulavasan replied:
For now we can use an inmemory hashmap instead instead of a KeyValueTable

awholegunch replied:
Can you write up what we can use as a replacement?

gokulavasan commented on tigon-docs/developer-guide/source/developer.rst:
```diff
@@ -0,0 +1,623 @@
+  @HashPartition("wordHash")
+  @ProcessInput("wordOut")
+  public void process(String word) {
+    this.wordCountsTable.increment(Bytes.toBytes(word), 1L);
```
Again here, no KeyValueTable.

gokulavasan commented on tigon-docs/developer-guide/source/developer.rst:
```diff
@@ -0,0 +1,623 @@
+
+
+The Need for Disabling Transactions
+-----------------------------------
```
There is no way to DisableTransaction in Tigon now. We need to remove this section.

gokulavasan commented on tigon-docs/developer-guide/source/developer.rst:
```diff
@@ -0,0 +1,623 @@
+    Check with Nitin
+
+Operations
+==========
```
I believe this is the CLI info for distributed mode.

gokulavasan commented on tigon-docs/developer-guide/source/examples.rst:
```diff
@@ -0,0 +1,75 @@
+
+- ``oauth.consumerKey``
+- ``oauth.consumerSecret``
+- ``oauth.Token``
```
oauth.accessToken
oauth.accessTokenSecret

gokulavasan commented on tigon-docs/developer-guide/source/faq.rst:
```diff
@@ -0,0 +1,142 @@
+   all of the description fields).
+#. Address all the review comments.
+#. After we review and accept your request, we’ll commit your code to the 
+   `cask/tigon <https://github.com/caskco/tigon>`__ repository.
```
caskdata/tigon

gokulavasan commented on tigon-docs/developer-guide/source/faq.rst:
```diff
@@ -0,0 +1,142 @@
+1. If you are planning a large change or contribution, discuss your plans on the `tigon-dev`
+   mailing list first.  This will help us understand your needs and best guide your solution in a
+   way that fits the project.
+#. Fork `Tigon <https://github.com/caskco/tigon>`__ into your own GitHub repository.
```
caskdata/tigon

gokulavasan commented on tigon-docs/developer-guide/source/getting-started.rst:
```diff
@@ -0,0 +1,93 @@
+========
+
+Pre-compiled sources and related files can be downloaded in a zip file: 
+`tigon-0.1.0.zip <http://cask.co/downloads/tigon/tigon-0.1.0.zip>`.
```
This link is not valid

gokulavasan commented on tigon-docs/developer-guide/source/tools.rst:
```diff
@@ -0,0 +1,74 @@
+   ``help``,Prints this helper text
+   ``version``,Prints the version
+   ``status``,Prints the current Tigon system status
+   ``exit``,Exits the shell
```
"quit" is the command to Quit the shell.

Commit:
Updates.

Commit:
Updates.

Commit:
Comment out Queues and Operations sections, and mark them as "to be completed".

Commit:
Updated to include link to existing example.

Commit:
Merge branch 'develop' into feature/docs-guides

Commit:
Remove tests from suggested build for Tigon.
Modify the build script to add a target to use pandoc to create rst files from the example README.md files.
Checked-in the converted files.
If pandoc is installed, the build script rebuilds from the latest READMEs; otherwise, the checked-in versions are used.
Modify the Sphinx config to exclude files in _includes directory.
Modified the examples.rst to include the README files rather than be a duplicate of their information.
Changed the CSS to have a bit more space around <pre> tags to allow for horizontal scrollbars on the bottom.

gokulavasan commented on tigon-docs/developer-guide/source/getting-started.rst:
```diff
@@ -0,0 +1,115 @@
+
+.. highlight:: console
+
+Tigon will not run with the default shared memory settings of the Macintosh, as 
```
TigonSQL will not run with default shared....

gokulavasan commented on tigon-docs/developer-guide/source/getting-started.rst:
```diff
@@ -0,0 +1,115 @@
+
+Tigon will not run with the default shared memory settings of the Macintosh, as 
+OS X does not provide sufficient shared memory regions by default. 
+To run Tigon, you'll need to revise these settings by executing these commands::
```
To run TigonSQL,..

Commit:
Add a test of the includes files; if they have changed, then copy the new versions in and use them.
Requires pandoc to work.

Commit:
Updates.

Commit:
Updates.

Commit:
Update.

Commit:
Updated copy of example rst file.

Commit:
Update.

Commit:
Update.

gokulavasan commented on tigon-docs/developer-guide/source/getting-started.rst:
```diff
@@ -0,0 +1,115 @@
+#. libcurl (if it is not already included in your operating system; see note following)
+#. libz (if it is not already included in your operating system; see note following)
+#. Apache Maven 3.0+ (required to build the example applications)
+#. In standalone mode, both Perl 5.x and Python 3.x are required
```
This should be rephrased as : If you are running TigonSQL StreamEngine in standalone mode, both Perl....

gokulavasan replied:
Oh I see there is a note below, then we can remove this point.

gokulavasan commented on tigon-docs/developer-guide/source/getting-started.rst:
```diff
@@ -0,0 +1,115 @@
+
+Pre-compiled sources and related files can be downloaded in a zip file: 
+`tigon-0.1.0.zip. 
+<http://repository.cask.co/downloads/co/cask/tigon/tigon-developer-release/0.1.0/tigon-developer-release-0.1.0.tgz>`__
```
I guess this link will be updated later? Or should it be done before merging to develop?

achalpandey commented on tigon-docs/developer-guide/source/_includes/sql-join-flow.rst:
```diff
@@ -0,0 +1,72 @@
+
+The only required property is:
+
+``http-port`` - The port to run the HTTP ingestion endpoints on;
```
Change `http-port` to `httpPort` to avoid confusion.

achalpandey commented on tigon-docs/developer-guide/source/_includes/sql-join-flow.rst:
```diff
@@ -0,0 +1,72 @@
+
+::
+
+    $ ./run_standalone.sh /path/to/SQLJoinFlow-<version>.jar co.cask.tigon.sqljoinflow.SQLJoinFlow <http-port>
```
How about we change this to 
$ ./run_standalone.sh /path/to/SQLJoinFlow-<version>.jar co.cask.tigon.sqljoinflow.SQLJoinFlow --httpPort=<portNumber>

I think we should make our instructions as simple as possible.

achalpandey commented on tigon-docs/developer-guide/source/_includes/twitter-analytics.rst:
```diff
@@ -0,0 +1,80 @@
+Overview
+--------
+
+An application collects Tweets and logs the top 10 hashtags used in the
```
An application THAT collects Tweets and ....

Commit:
Updates as per comments.

awholegunch commented:
I've updated the PR. Please take another look.

gokulavasan commented on tigon-docs/developer-guide/source/architecture.rst:
```diff
@@ -0,0 +1,492 @@
+implemented using in-memory data structures. 
+
+In Distributed Mode, each Flowlet is a YARN container, and Queues are implemented using
+HBase Tables. In Distributed Mode an applications can, if required, persist data to HBase.
```
In Distributed Mode, an application can, if required, persist, persist...

gokulavasan commented on tigon-docs/developer-guide/source/architecture.rst:
```diff
@@ -0,0 +1,492 @@
+HBase Tables. In Distributed Mode an applications can, if required, persist data to HBase.
+Since multiple Flowlets can potentially be accessing the same Tables, it's advised that an
+application use Cask Tephra to write to the HBase Tables transactionally. (Details of this
+are covered in our `Developer Guide. <developer.html>`__)
```
Can we point the users to the exact location of this info in the developer guide instead of the main landing page of developer guide?

gokulavasan commented on tigon-docs/developer-guide/source/architecture.rst:
```diff
@@ -0,0 +1,492 @@
+Currently the number of instances of an ``AbstractInputFlowlet`` is limited to one. Other
+flowlets don’t have this constraint. ``AbstractInputFlowlet`` exposes HTTP and TCP
+endpoints for the user to ingest data. Details on how to use these endpoints is covered in
+our `Developer Guide. <developer.html>`__
```
Same here. Point them to the exact location?

gokulavasan commented on tigon-docs/developer-guide/source/architecture.rst:
```diff
@@ -0,0 +1,492 @@
+
+Designing a high-performance Data Stream Management System (DSMS) requires solving a large
+number of unique technical challenges stemming from extreme data rates and real-time
+processing requirements. In this document, we present the architecture of the Tigon stream
```
the architecture of TigonSQL stream database...

gokulavasan replied:
Or probably remove database reference and use DSMS

gokulavasan commented on tigon-docs/developer-guide/source/architecture.rst:
```diff
@@ -0,0 +1,492 @@
+number of unique technical challenges stemming from extreme data rates and real-time
+processing requirements. In this document, we present the architecture of the Tigon stream
+database specifically designed to address these challenges. Tigon was developed by AT&T
+Labs-Research and Cask Data, Inc. for monitoring high-rate network data streams and is
```
Hmm.. this needs to be rephrased. TigonSQL Stream Database was developed by AT&T Labs-Research for monitoring ...

gokulavasan commented on tigon-docs/developer-guide/source/architecture.rst:
```diff
@@ -0,0 +1,492 @@
+currently used as a vehicle for networking and streaming research by AT&T.
+
+Even though the system is specialized for network monitoring applications, most of the
+Tigon design principles apply to any streaming application required to process high-rate
```
most of the design principles apply...

gokulavasan commented on tigon-docs/developer-guide/source/architecture.rst:
```diff
@@ -0,0 +1,492 @@
+
+The document is organized as:
+
+- An overview of the `stream query language`_ used by Tigon and a description of the main types 
```
TigonSQL.. I think any reference to Tigon in "TigonSQL Architecture" section should be replaced with TigonSQL

gokulavasan commented on tigon-docs/developer-guide/source/developer.rst:
```diff
@@ -0,0 +1,752 @@
+
+The potential problem with this is that the first two instances might
+both attempt to increment the counter for the word *scream* at the same time,
+leading to a write conflict. To avoid conflicts, we can use hash-based partitioning::
```
I think this part has not been updated. Refer to the source.

gokulavasan commented on tigon-docs/developer-guide/source/developer.rst:
```diff
@@ -0,0 +1,752 @@
+this is implemented through in-memory data structures. In Distributed Mode, it is
+implemented using HBase Tables. This provides reliability and fault-tolerance to the Flow
+system such that when a Flowlet instances dies, it is respawned and it starts reading
+events from the queues from the next event in the queue.
```
and it starts reading from the next event in the queue.

gokulavasan commented on tigon-docs/developer-guide/source/developer.rst:
```diff
@@ -0,0 +1,752 @@
+-----------------------------------------------
+
+Tigon internally uses Tephra extensively to complete transactional operations. Tephra can
+also be leveraged by developers to write to HBase transactionally. To do this, wrap an
```
@ParthGandhi Should we mention that data flow between Queues are stored in HBase transactionally and is transparent to the user. But if the user wants to persist data in his application logic, we can use Tephra to get ACID properties?

ParthGandhi replied:
If you want to add that, it should be under a separate section - This is specifically for how and what a user does to write transactionally. If you want to discuss internals that are transparent to the user it should go under `Architecture` or similar.

gokulavasan commented on tigon-docs/developer-guide/source/developer.rst:
```diff
@@ -0,0 +1,752 @@
+aggregation, and joins of Streams. This can be highly useful for use cases where a large
+ingestion rate is required. 
+
+However, it must be noted that as the data in TigonSQL is held in-memory, there is a
```
is held in-memory and thus there is a possibility of data loss ...

gokulavasan commented on tigon-docs/developer-guide/source/developer.rst:
```diff
@@ -0,0 +1,752 @@
+
+In order to use the TigonSQL library in your flow, you need a Flowlet that extends
+``AbstractInputFlowlet``. To use the StreamEngine, implement the ``create method``. The
+building blocks of the StreamEngine are the ``StreamSchema``, ``addJSONInput``, and
```
StreamSchema objects, addJSONInput and addQuery methods.

gokulavasan commented on tigon-docs/developer-guide/source/developer.rst:
```diff
@@ -0,0 +1,752 @@
+
+You can ingest data through the HTTP Port using a curl command such as::
+
+  curl -v -X POST http://localhost:63541/v1/tigon/<InputName> -d '{ "data" : [ “12495”, “233“ ] }’
```
locahost:< port >/v1/tigon/< InputName >

gokulavasan commented on tigon-docs/developer-guide/source/developer.rst:
```diff
@@ -0,0 +1,752 @@
+You can choose to ingest data through either HTTP or TCP endpoints; in the case above, the TCP server is
+running on 63537. There is one TCP endpoint for each input stream.
+
+If the Flow is running in Distributed Mode on a cluster, you can use discover using
```
you can use the discover and service info commands to find out the endpoints.

gokulavasan commented on tigon-docs/developer-guide/source/developer.rst:
```diff
@@ -0,0 +1,752 @@
+        .withFlowlets().add(new Tokenizer())
+                       .add(new WordsFilter("the"))
+                       .add(new WordsCounter())
+        .connect().fromStream("text").to("Tokenizer")
```
remove -  " .fromStream("text").to("Tokenizer")  "

gokulavasan commented on tigon-docs/developer-guide/source/releasenotes.rst:
```diff
@@ -0,0 +1,33 @@
+  as described in the `Getting Started with Tigon guide. <getting-started.html#macintosh-os-x>`__
+- The maximum number of ``AbstractInputFlowlet`` instances is limited to one.
+- Metrics are currently not supported.
+- A Flow needs to have at least two Flowlets.
```
This issue can be removed.

gokulavasan commented:
@awholegunch Few comments. Take a look. Thanks!

Commit:
Updates based on comments.

gokulavasan commented on tigon-docs/developer-guide/source/getting-started.rst:
```diff
@@ -0,0 +1,120 @@
+========
+
+Pre-compiled sources and related files can be downloaded in a zip file: 
+`tigon-0.1.0.zip. 
```
tigon-0.2.0.zip

Commit:
Merge branch 'develop' into feature/docs-guides

Commit:
Correct filename.

Commit:
March the README.md file.

Commit:
Fix broken links to APIs.

Commit:
Adding description of the Maven archetype.

Commit:
Fix retrieval of build version to use grep instead of maven.

gokulavasan commented on tigon-docs/developer-guide/source/getting-started.rst:
```diff
@@ -0,0 +1,120 @@
+
+  git clone https://github.com/caskdata/tigon.git
+  cd tigon
+  mvn clean package -P sql-lib -DskipTests 
```
mvn clean package -DskipTests -Pdist

We don't want them to rebuild sql-lib unless they want to. Lets cover that in the "Problems with the Downloaded Package" section.

gokulavasan commented on tigon-docs/developer-guide/source/getting-started.rst:
```diff
@@ -0,0 +1,120 @@
+---------------------------------------
+
+If the downloaded package doesn't work, try to build a tarball for your system
+as described above.
```
If you face problems using TigonSQL, you can build the TigonSQL streaming library from the git repo. 

mvn clean package install -DskipTests -P sql-lib -Pdist 

The TigonSQL jar will be installed locally and this will be used by maven to create Tigon applications when they are created on the same machine.

@chtyim  Does this sound ok?

chtyim replied:
you don't need package install, just mvn clean install

chtyim replied:
Also, you can use comma to specifies multiple profiles like:

`mvn clean install -DskipTests -P sql-lib,dist`

chtyim replied:
The rest sounds good to me.

gokulavasan commented on tigon-docs/developer-guide/source/architecture.rst:
```diff
@@ -0,0 +1,494 @@
+   :width: 600px
+
+
+TigonSQL is a library built for Tigon. It contains a In-memory Stream Processing engine
```
an In-memory Stream Processing Engine...?

gokulavasan commented on tigon-docs/developer-guide/source/developer.rst:
```diff
@@ -0,0 +1,755 @@
+Flowlets and Instances
+----------------------
+You can have one or more instances of any given Flowlet, each consuming a disjoint
+partition of each input. You can control the number of instances programmatically via the
```
You can control the number of instances through the CLI in Distributed Mode. 

No REST interfaces or CDAP Console in Tigon.

Commit:
Updates.

Commit:
Fixes for scrolling and formatting problems in HTML.

awholegunch commented:
Updated to fix comments. Please review again.

Commit:
Cleanup formatting.

Commit:
Merge branch 'develop' into feature/docs-guides

Conflicts:
	tigon-examples/SQLJoinFlow/README.md
	tigon-examples/TwitterAnalytics/README.md

Commit:
Fix example READMEs after merge, fixing formatting, link and correcting edits.
Check-in cached files.
Fix the conf.py to not call the get_sdk_version method more than once.
Fix typo in examples.rst.

awholegunch commented:
Updated staging serve; now at http://stg-web101.sw.joyent.continuuity.net/tigon/0.2.0-SNAPSHOT/en/
Fixed merge conflict and made edits. Please check again.

gokulavasan commented on tigon-docs/developer-guide/source/getting-started.rst:
```diff
@@ -0,0 +1,144 @@
+========
+
+Pre-compiled sources and related files can be downloaded in a zip file: 
+`tigon-developer-release-0.2.0.zip. 
```
We provide a tgz package, not a zip.

gokulavasan commented on tigon-docs/developer-guide/source/getting-started.rst:
```diff
@@ -0,0 +1,144 @@
+After the build completes, you will have a distribution of Tigon under the
+`tigon-distribution/target/` directory.  
+
+Take the `tigon-<version>.tar.gz` file and unzip it into a suitable location.
```
tigon-sdk-<version>.tar.gz

Commit:
Fix references to archive files.
Adjust highlighting of table of contents.
Add scroll highlighting.

Commit:
Fix reference to tar file.

gokulavasan commented:
LGTM :+1:
